test/core.spec.js

```diff
@@ -1,6 +1,6 @@
 const expect = require('expect.js');
 
-const HTMLHint = require('../dist/htmlhint.js').hint;
+const HTMLHint = require('../dist/htmlhint.js').HTMLHint;
```

Comment:
I personally dont like all this changes whithin **this** PR  
I like clean PR that makes one specific change to the codebase or add one new feature

But this PR seems to tackle the GitHub Actions Codecov job/step **AND** some refactoring to the codebase

Bumping up some needed newly dependencies and `package.json` changes may be ok within this PR

Reply:
Yes, I agree. Gonna separate these 2 things.